app/src/main/java/ui/MapActivity.kt

```diff
@@ -327,25 +325,29 @@ class MapActivity : DrawerActivity(), OnMapReadyCallback, GoogleMap.OnMarkerClic
         map?.setOnMarkerClickListener(this)
 
         doAsync {
-            val places = placeDao.getPlaces()
+            val thePlaces = placeDao.getPlaces()
             val placeId = intent.getIntExtra(Constants.ID, -1)
             var theMarker: Marker? = null
-            places.mapNotNullTo(this@MapActivity.places) {
+            thePlaces.mapNotNullTo(places) {
                 // Create a marker for this
                 val marker = map?.addMarker(
                     MarkerOptions()
                         .position(it.coordinates)
                         .draggable(false)
                         .visible(true)
-                ) ?: return@mapNotNullTo null
+                )
+
+                if (marker == null) {
```

Comment:
Was it recommended to do this vs the previous implementation?

It also looks like you should check that `map` isn't null first, as everything else is null otherwise.
If you want, you can also get all the places first, then do
`val theMarker = places.firstOrNull { it.first.id == placeId }` to avoid your own loop and null check.

Reply:
The recommendation is to not use labelled expressions. I didn't like how this code is formatted in any case. I like that recommendation, I'll switch it around.